### PR TITLE
Add a test for a custom promise using the 'handle' attribute

### PR DESCRIPTION
**Pull request by vpodzime**
Just making sure the '$(this.handle)' expression is properly
expanded to the value of the attribute.

Ticket: CFE-3439
Changelog: None

Merge together:
https://github.com/cfengine/core/pull/4795
https://github.com/cfengine/masterfiles/pull/2087

### CODE REVIEW

**Comment by vpodzime** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7145/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7145/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7145/

**Review by olehermanse — APPROVED** [review]

File: tests/acceptance/30_custom_promise_types/11_handle.cf [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+bundle agent test
+{
+  meta:
+      "description" -> { "CFE-3439" }
+        string => "Test a custom promise using the 'handle' attribute";
+
+  multiline_insert:
+      "$(G.testfile)"
+        lines => { "handle is:", "$(this.handle)" },
+        handle => "example_handle";
```
Comment (olehermanse) [inline tests/acceptance/30_custom_promise_types/11_handle.cf:53]: Every time I've seen handle used, it's for another promise to use in `depends_on`, so I would do that here or add another test. You could say that then you are testing depends_on as well, but I would say without it you are not testing the most important part of handle.

```suggestion
  multiline_insert:
      "$(G.testfile)"
        lines => { "handle is:", "$(this.handle)" },
        handle => "example_handle_1";
  classes:
    "found_example_handle_1" # Should be defined
      depends_on => "example_handle_1";
    "found_example_handle_2" # Shouldn't be defined
      depends_on => "example_handle_2";
```

Comment (vpodzime) [reply tests/acceptance/30_custom_promise_types/11_handle.cf:53]: > Every time I've seen handle used, it's for another promise to use in `depends_on`, so I would do that here or add another test

It's also being used to trick the promise locking, but that's not implemented for custom promises at all.

**Pull request merged**